### ISSUE TITLE
Request is being made for sim version that doesn't exist on website

### ISSUE THREAD

**Issue by liammulh** [phetsims/rosetta#374]
I was checking Rosetta's logs this morning to see if there were any errors, and I saw 5 instances of:

```
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: initiating build request for sim: energy-skate-park, locale: pl
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: getting sim metadata
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: returning sim metadata mutex
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: old saved pl/energy-skate-park translation(s) with same user id extant; deleting them
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: using cached sim metadata
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: returning sim metadata
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: latest version of the sim: 1.1.27.
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: getting dependencies from https://phet.colorado.edu/sims/html/energy-skate-park/1.1.27/dependencies.json
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: done attempting to delete pl/energy-skate-park translation(s) in short-term storage
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: error: Request failed with status code 404 {"config":{"headers":{"Accept":"application/json, text/plain, */*","User-Agent":"axios/0.24.0"},"maxBodyLength":-1,"maxContentLength":-1,"method":"get","timeout":0,"transformRequest":[null],"transformResponse":[null],"transitional":{"clarifyTimeoutError":false,"forcedJSONParsing":true,"silentJSONParsing":true},"url":"https://phet.colorado.edu/sims/html/energy-skate-park/1.1.27/dependencies.json","xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN"},"status":404}
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: returning dependencies
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: getting phet-io metadata request with https://phet.colorado.edu/services/metadata/phetio?&active=true&latest=true
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: build request object:
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info:     api: 2.0
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info:     simName: energy-skate-park
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info:     version: 1.1.27
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info:     locales: pl
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info:     servers: production
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info:     brands: phet
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info:     translatorId: 196556
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: info: sending build request to https://phet.colorado.edu/deploy-html-simulation
Feb 23 02:11:27 phet-server2.int.colorado.edu rosetta[375109]: error: build request result: Error: Request failed with status code 500
```

There were 4 (possibly red herring) instances of:

```
Feb 23 02:21:02 phet-server2.int.colorado.edu rosetta[375109]: error: long term storage of pl/energy-skate-park failed
```

Build server logs:

```
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: api:"2.0"
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: dependencies:"error: unable to get dependencies"
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: simName:"energy-skate-park"
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: version:"1.1.27"
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: locales:["pl"]
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: servers:["production"]
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: brands:["phet"]
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: translatorId:196556
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: authorizationCode:"d!8-A3fBvG2Jd#HV^SkW"
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]: SyntaxError: Unexpected token e in JSON at position 0
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at JSON.parse (<anonymous>)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at postQueueDeploy (/data/share/phet/build-server/perennial/js/build-server/build-server.js:111:24)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at Layer.handle [as handle_request] (/data/share/phet/build-server/perennial/node_modules/express/lib/router/layer.js:95:5)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at next (/data/share/phet/build-server/perennial/node_modules/express/lib/router/route.js:137:13)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at Route.dispatch (/data/share/phet/build-server/perennial/node_modules/express/lib/router/route.js:112:3)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at Layer.handle [as handle_request] (/data/share/phet/build-server/perennial/node_modules/express/lib/router/layer.js:95:5)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at /data/share/phet/build-server/perennial/node_modules/express/lib/router/index.js:281:22
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at Function.process_params (/data/share/phet/build-server/perennial/node_modules/express/lib/router/index.js:335:12)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at next (/data/share/phet/build-server/perennial/node_modules/express/lib/router/index.js:275:10)
Feb 23 02:08:34 phet-server2.int.colorado.edu build-server[1548523]:     at /data/share/phet/build-server/perennial/node_modules/body-parser/lib/read.js:130:5
Feb 23 02:10:02 phet-server2.int.colorado.edu build-server[1548523]: info: deploy request received, original URL = http://phet.colorado.edu/deploy-html-simulation
```

On the website:

<img width="631" alt="esp-index" src="https://user-images.githubusercontent.com/60749003/220967737-bf964d9f-134b-4b5b-b094-c3e911ee7ca4.png">

<hr/>

It looks like Rosetta thinks the latest version of Energy Skate Park is 1.1.27, but evidently the latest version is 1.2.2. Rosetta is choking when requesting the dependencies JSON file.

**Comment by liammulh** [comment]
This code I wrote is confusing me:

https://github.com/phetsims/rosetta/blob/79b01642fda4e2b4d1633061c79611959cf21b8b/src/server/translationApi/getSimVersionObject.js#L18-L34

I have no idea why this function is named `getSimVersionObject`. It seemingly returns a string. When we call this function, we expect it to be an object with a `string` key:

https://github.com/phetsims/rosetta/blob/79b01642fda4e2b4d1633061c79611959cf21b8b/src/server/translationApi/requestBuild.js#L29-L33

This is highly disturbing. I am not sure how this is working. TypeScript really would have saved my ass here.

**Comment by liammulh** [comment]
- [x] check logs to ensure `latest version of the sim:` logs have a version
- [ ] check sim metadata to see what `projects[ i ].version` is (a string or an object?)

**Comment by liammulh** [comment]
I've verified that the `getSimVersionObject` function is behaving properly. So now the question becomes why is the version for Energy Skate Park 1.1.27?

In the sim metadata, the version for Energy Skate Park is 1.2.2:

```
        {
            "visible": true,
            "name": "html/energy-skate-park",
            "id": 198,
            "type": 2,
            "version": {
                "string": "1.2.2",
                "major": 1,
                "minor": 2,
                "dev": 2,
                "timestamp": 0
            },
```

**Comment by liammulh** [comment]
I set this up in my local instance of Rosetta:

```js
const requestBuild = async ( simName, locale, userID ) => {

  logger.info( `initiating build request for sim: ${simName}, locale: ${locale}` );
  const simVersionObject = await getSimVersionObject( simName );

  console.log( `----------> simVersionObject = ${JSON.stringify( simVersionObject, null, 4 )}` );

  logger.info( `latest version of the sim: ${simVersionObject.string}.` );
  const dependencies = await getDependencies( simName, simVersionObject.string );
```

This is the output:

```
info: initiating build request for sim: energy-skate-park, locale: ab
info: getting sim metadata
info: returning sim metadata mutex
info: old saved ab/energy-skate-park translation(s) with same user id extant; deleting them
info: using cached sim metadata
info: returning sim metadata
----------> simVersionObject = {
    "string": "1.1.27",
    "major": 1,
    "minor": 1,
    "dev": 27,
    "timestamp": 0
}
info: latest version of the sim: 1.1.27.
info: getting dependencies from https://phet.colorado.edu/sims/html/energy-skate-park/1.1.27/dependencies.json
info: done attempting to delete ab/energy-skate-park translation(s) in short-term storage
error: Request failed with status code 404 {"config":{"headers":{"Accept":"application/json, text/plain, */*","User-Agent":"axios/0.24.0"},"maxBodyLength":-1,"maxContentLength":-1,"method":"get","timeout":0,"transformRequest":[null],"transformResponse":[null],"transitional":{"clarifyTimeoutError":false,"forcedJSONParsing":true,"silentJSONParsing":true},"url":"https://phet.colorado.edu/sims/html/energy-skate-park/1.1.27/dependencies.json","xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN"},"status":404}
info: returning dependencies
info: getting phet-io metadata request with https://phet.colorado.edu/services/metadata/phetio?&active=true&latest=true
info: build request object:
info:     api: 2.0
info:     simName: energy-skate-park
info:     version: 1.1.27
info:     locales: ab
info:     servers: production
info:     brands: phet
info:     translatorId: 136803
warn: send build request flag is false; check your config
```

**Comment by liammulh** [comment]
<details>

<summary>Patch</summary>

```
diff --git a/src/server/translationApi/getSimVersionObject.js b/src/server/translationApi/getSimVersionObject.js
index 15dabf0..78f6aaf 100644
--- a/src/server/translationApi/getSimVersionObject.js
+++ b/src/server/translationApi/getSimVersionObject.js
@@ -13,7 +13,7 @@ import logger from './logger.js';
  * Extract the version object for the given sim from the sim metadata and return it.
  *
  * @param simName - name of the sim
- * @returns {Promise<string>}
+ * @returns {Promise<Object|String>}
  */
 const getSimVersionObject = async simName => {
   let latestVersionOfSim = '';
@@ -21,7 +21,7 @@ const getSimVersionObject = async simName => {
     const simMetadata = await getSimMetadata();
     const projects = simMetadata.projects;
     for ( const i of Object.keys( projects ) ) {
-      if ( projects[ i ].name.includes( simName ) ) {
+      if ( projects[ i ].name === `html/${simName}` ) {
         latestVersionOfSim = projects[ i ].version;
       }
     }
```

</details>

**Comment by liammulh** [comment]
I need to test the above patch.

- [ ] configure to use tests branch of babel
- [ ] configure to perform string commits
- [ ] configure to not send build requests
- [ ] "publish" changes to ab/energy-skate-park and ensure you see 1.2.2 for the version
- [ ] "publish" changes to ab/energy-skate-park-basics and ensure you see 1.1.27 for the version
- [ ] note version of two random sims here
  - sim A:
  - sim B: 
- [ ] "publish" changes to sim A and ensure you see the correct version
- [ ] "publish" changes to sim B and ensure you see the correct version

**Comment by liammulh** [comment]
@oliver-phet, please close this if the translator says it fixed their problem. Otherwise please comment with their feedback.

If we don't hear back from the translator, I will close once we are reasonably confident this is no longer an issue.

**Comment by liammulh** [comment]
I labeled this fixed-awaiting-deploy, but it's actually deployed. I couldn't find a better tag.

**Comment by oliver-phet** [comment]
From _pl translator:

> Unfortunately, the problems persist.
> 1. /pl/acid-base-solutions
> I filled in the blank strigs, saved and published. Next to the title there was information in parentheses about processing. After a few hours, the data was updated.
> 
> 2. /pl/area-builder
> Just in case, only after publishing changes in the previous application did I complete this one ( by the way, part of it was already translated in the old Translation Utility, now it's disappeared). I saved it and published. There were again messages about saving and publishing (within half an hour) changes. This time, on the main page next to the title, there was NO information about processing. After a few hours I checked - the completed strings were empty - I completed and saved again, but this time it was impossible to publish. An error message appears.

**Comment by liammulh** [comment]
@oliver-phet when was the email sent?

**Comment by oliver-phet** [comment]
> @oliver-phet when was the email sent?

Came into phethelp yesterday Feb 26, 2023, 9:59 AM

**Comment by liammulh** [comment]
The last commit in Babel for pl/acid-base-solutions was:

> /pl/acid-base-solutions

> Next to the title there was information in parentheses about processing.

I assume they mean the "(pending)" note. This is expected.

> After a few hours, the data was updated.

By data, I assume they mean the string translation statistics.

I am confused. What is the issue? Were they expecting the data to be updated sooner? Did they see the "(pending)" note go away before the data updated?

Also, the last commit to Babel for pl/acid-base-solutions was one day before I deployed this fix (Feb 22):

```
57423d376	phet-dev	Wed Feb 22 08:34:35 2023 -0700	automated commit from rosetta for sim/lib acid-base-solutions, locale pl
```

> /pl/area-builder

When the email was sent (Feb 26), the last commit to Babel for pl/area-builder was also one day before I deployed this fix (Feb 22):

```
08f119e34	phet-dev	Wed Feb 22 08:42:53 2023 -0700	automated commit from rosetta for sim/lib area-builder, locale pl
```

The most recent commit to pl/area-builder was today (Feb 27):

```
b025f5ecd	phet-dev	Mon Feb 27 06:09:41 2023 -0700	automated commit from rosetta for sim/lib area-builder, locale pl
```

@oliver-phet, can you send a follow-up email asking if they saw the same issue with area-builder with their most recent translation (the one done on the 27th)?

**Comment by oliver-phet** [comment]
> All titles and many other strings was in English, but now, as I see, is ok :) It's just that in the meantime I re-entered the translations.
> But in Area Model Algebra information is still 67% (59 of 88)

I think this translators confusion is primarily with the translation statistics. How quickly do we expect those to update? It might be good to indicate that delay in the user guide.

**Comment by liammulh** [comment]
> I think this translators confusion is primarily with the translation statistics. How quickly do we expect those to update? It might be good to indicate that delay in the user guide.

This is complicated. It depends on several factors, some of which the translation utility doesn't have control over. I don't think it would be a good idea to put a number in the user guide right now. I will open up a separate issue for this.

**Comment by liammulh** [comment]
Issue here: https://github.com/phetsims/rosetta/issues/379.
